custom/icds_reports/management/commands/rebuild_ccs_record_ucr.py

```diff
@@ -9,31 +9,41 @@
 
 
 class Command(BaseCommand):
-    help = "Rebuild ccs record cases ucr for pnc_complete"
+    help = "Rebuild open ccs record cases and their corresponding person cases "
 
-    def handle(self):
-        indicator_config_id = 'static-ccs_record_cases'
-        case_type_of_xmlns = 'ccs_record'
+    def get_asdfsd(self, indicator_config_id, case_type_of_xmlns):
         config = _get_config_by_id(indicator_config_id)
-
         document_store = get_document_store_for_doc_type(
             config.domain, config.referenced_doc_type,
             case_type_or_xmlns=case_type_of_xmlns,
             load_source="build_ccs_record_ucr",
         )
+        return config, document_store
+
+    def handle(self):
+        ccs_record_config, ccs_record_document_store = self.get_asdfsd('static-ccs_record_cases', 'ccs_record')
+        person_config, person_document_store = self.get_asdfsd('static-person_cases_v3', 'person')
 
         current_month_start = datetime.date.today().replace(day=1)
         last_month_start = (current_month_start - datetime.timedelta(days=1)).replace(day=1)
 
         current_month_start = current_month_start.strftime('%Y-%m-%d')
         last_month_start = last_month_start.strftime('%Y-%m-%d')
-        current_month_doc_ids = CcsRecordMonthly.objects.filter(pnc_complete=1,
-                                                                month=current_month_start).values('case_id')
-        docs_last_month = CcsRecordMonthly.objects.filter(pnc_complete=1,
-                                                          month=last_month_start).values('case_id')
+        current_month_doc_ids = CcsRecordMonthly.objects.filter(open_in_month=1,
+                                                                month=current_month_start).values('case_id', 'person_case_id')
```

Comment:
Do we know approximately how many rows this is? Any reason to worry about how long this will take / how big the result set will be?

Reply:
There are 5731624 cases in this. 
This would have the Cost about this:
```
icds_ucr=# EXPLAIN select case_id,person_case_id from ccs_record_monthly where month='2019-10-01' and open_in_month=1;

                                                           QUERY PLAN                                                            
---------------------------------------------------------------------------------------------------------------------------------
 Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
   Task Count: 64
   Tasks Shown: One of 64
   ->  Task
         Node: host=100.71.184.232 port=5432 dbname=icds_ucr
         ->  Gather  (cost=1000.00..111228.10 rows=81660 width=74)
               Workers Planned: 5
               ->  Parallel Seq Scan on ccs_record_monthly_102712 ccs_record_monthly  (cost=0.00..102062.10 rows=16332 width=74)
                     Filter: ((month = '2019-10-01'::date) AND (open_in_month = 1))
(9 rows)
```

We would need to time it properly Like in the evening but not near the aggregation. Also, based on comparison with Ap data pull [here](https://github.com/dimagi/commcare-hq/pull/25889/files#diff-bd935ad77fb44cf20302daf8582b3e08). It would be much lesser cost.
So, I think if we dont run it during day, and not near aggregation like 6-7PM It should be fine. .
@emord  what do you think?

Reply:
Recently, I think I've seen quite a few of these queries that require a `month` filter without a `supervisor_id` filter. It may be worth adding a `month` index on this table so that these are faster in the future. 

You could potentially break this query up into smaller queries based on supervisor_id, but because this also needs the `person_case_id` I don't really have a better suggestion or table for this at  the moment.

Reply:
I believe now the cost should be reduced now I have changed the code to pull cases in the chunk of supervisor_id. Does this look good?